block/slvalidator.go

```diff
@@ -146,6 +146,11 @@ func (v *SettlementValidator) ValidateDaBlocks(slBatch *settlement.ResultRetriev
 		return types.NewErrStateUpdateNumBlocksNotMatchingFraud(slBatch.EndHeight, numSLBlocks, numSLBlocks)
 	}
 
+	currentProposer := v.blockManager.State.GetProposer()
+	if currentProposer == nil {
```

Comment:
will it allow historical sync of a halted rollapp? (rollapp without active proposer)

Reply:
as we're getting this from the state, and the state should reflect the rollapp proposer for height (it's not currently but there is an open issue on it), it means that by the time we've got here the chain is most likely halted (i.e has no current proposer).

in that case we have nothing to validate.